openedx/core/djangoapps/oauth_dispatch/dot_overrides/views.py

```diff
@@ -0,0 +1,79 @@
+"""
+Classes that override default django-oauth-toolkit behavior
+"""
+from __future__ import unicode_literals
+
+from oauth2_provider.exceptions import OAuthToolkitError
+from oauth2_provider.http import HttpResponseUriRedirect
+from oauth2_provider.models import get_application_model
+from oauth2_provider.scopes import get_scopes_backend
+from oauth2_provider.settings import oauth2_settings
+from oauth2_provider.views import AuthorizationView
+
+
+# TODO (ARCH-xxx) remove once we have full support of OAuth Scopes
+class EdxOAuth2AuthorizationView(AuthorizationView):
+    """
+    Override the AuthorizationView's GET method so the user isn't
+    prompted to approve the application if they have already in
+    the past, even if their access token is expired.
+
+    This is a temporary override of the default implementation
+    in order to accommodate our Restricted Applications support
+    until OAuth Scopes are fully supported.
+    """
+    def get(self, request, *args, **kwargs):
+        try:
+            require_approval = request.GET.get(
+                "approval_prompt",
+                oauth2_settings.REQUEST_APPROVAL_PROMPT,
+            )
+            if require_approval != 'auto_even_if_expired':
+                return super(EdxOAuth2AuthorizationView, self).get(request, *args, **kwargs)
+
+            scopes, credentials = self.validate_authorization_request(request)
+            all_scopes = get_scopes_backend().get_all_scopes()
+            kwargs["scopes_descriptions"] = [all_scopes[scope] for scope in scopes]
+            kwargs['scopes'] = scopes
+
+            # at this point we know an Application instance with such client_id exists in the database
+            application = get_application_model().objects.get(client_id=credentials['client_id'])
+            kwargs['application'] = application
+            kwargs['client_id'] = credentials['client_id']
+            kwargs['redirect_uri'] = credentials['redirect_uri']
+            kwargs['response_type'] = credentials['response_type']
+            kwargs['state'] = credentials['state']
+
+            self.oauth2_data = kwargs
+            # following two loc are here only because of https://code.djangoproject.com/ticket/17795
+            form = self.get_form(self.get_form_class())
+            kwargs['form'] = form
```

Comment:
do you think it would be better to keep the copied code as is - without doing any refactoring (even Clean code type) updates?  If so, I can add a comment to explain that it's a close copy of the original base class' implementation.

Reply:
I added docstring and text to explain that this code was copied as-is, except for a few modifications that we highlighted.